src/aleph/sdk/chains/substrate.py

```diff
@@ -50,6 +76,31 @@ def verify_signature(
     signature: Union[bytes, str],
     public_key: Union[bytes, str],
     message: Union[bytes, str],
-) -> bool:
-    """TODO: Implement this"""
-    raise NotImplementedError("Not implemented yet")
+) -> None:
+    if isinstance(signature, str):
+        if signature.startswith("0x"):
+            signature = signature[2:]
+        signature = bytes.fromhex(signature)
+    if isinstance(public_key, str):
+        if public_key.startswith("0x"):
+            public_key = public_key[2:]
+        public_key = bytes.fromhex(public_key)
+    if isinstance(message, str):
+        message = message.encode()
+
+    try:
+        verified = verify(signature, message, public_key)
+        if not verified:
+            # Another attempt with the data wrapped, as discussed in https://github.com/polkadot-js/extension/pull/743
+            verified = verify(signature, b"<Bytes>" + message + b"</Bytes>", public_key)
+        if not verified:
+            raise BadSignatureError
```

Comment:
Could this be written in a smarter way using `or` ?

```suggestion
        # Two different attempts at verifying the data wrapped, as discussed in https://github.com/polkadot-js/extension/pull/743
        if not (
            verify(signature, message, public_key) 
            or verify(signature, b"<Bytes>" + message + b"</Bytes>", public_key)
        ):
            raise BadSignatureError
```